arkouda/segarray.py

```diff
@@ -361,6 +361,9 @@ def __eq__(self, other):
             eq[leneq] = intersection
         return eq
 
+    def __len__(self) -> int:
+        return self.segments.size
```

Comment:
Should we not just return `self.size` here? It is set from `segments.size`, so the 2 should be equivalent.

Reply:
yah that's a better idea!